src/color.rs

```diff
@@ -316,19 +316,19 @@ fn detect_color(conin: &mut dyn ConsoleRead, color: u16) -> io::Result<bool> {
     let mut total_read = 0;
 
     let timeout = Duration::from_millis(CONTROL_SEQUENCE_TIMEOUT);
-    let now = SystemTime::now();
     let bell = 7u8;
 
     // Either consume all data up to bell or wait for a timeout.
-    while buf[0] != bell && now.elapsed().unwrap() < timeout {
-        match conin.read(&mut buf) {
-            Ok(b) => total_read += b,
-            // Don't error out on a would block- this just means no response since async.
-            Err(err) if err.kind() == io::ErrorKind::WouldBlock => {}
-            Err(err) => return Err(err),
+    if conin.poll_timeout(timeout) {
+        while buf[0] != bell {
+            match conin.read(&mut buf) {
+                Ok(b) => total_read += b,
+                // Don't error out on a would block- this just means no response since async.
+                Err(err) if err.kind() == io::ErrorKind::WouldBlock => {}
```

Comment:
I think this line (and it's comment) should go away.  Now that you are polling conin then this should not happen.  Maybe it could with some sort of interrupt to the underlying select (but I don't think poll will return true then anyway) but even so I think just falling through to the general error is better.  If for some unknown reason this did happen we might wind up blocking when the while loop repeat and calls read without the poll.